### PR TITLE
Fix/improve a11y everyone can be bored

### PR DESCRIPTION
**Pull request by sandrina-p**
Hey there 👋 

Really cool web app, I added to my iPhone 5 homescreen!
On Desktop I use a lot the keyboard and noticed the navigation was impossible, so I improved the Accessibility overall, for Screen Readers and Keyboard navigation:

- Fix headings usage (h1 <-> h2)
- Add missing headings for SR
- Make buttons real `<button />` tag
- Add missing hover/focus styling to buttons
- Group messages in a `<ul />` list
- Make Conversation live with SR (`aria-live` attr).

As bonus I also:
- Fixed the message timestamp message to stay the same over time (as each render it was creating a new Date)
- Improve a little big how the site looks on small screens (320) by reducing some paddings.
- Added cool stuff do when bored

### CODE REVIEW

File: src/app.js [hunk 1]
```diff
@@ -77,15 +83,15 @@ export default () => {
                   {done ? (
                     'Amazing'
                   ) : (
-                    <span role="img" aria-label="Yes">
```
Comment (sandrina-p) [inline src/app.js:80]: @SaraVieira I'm curious... why using `role="img"` here?

Comment (SaraVieira) [reply src/app.js:80]: YES! 
It's so screen readers don't try to read the Unicode inside of it and only read the aria-label
https://tink.uk/accessible-emoji/

File: src/components/Elements.js [hunk 2]
```diff
@@ -105,3 +111,14 @@ export const ChatContainer = styled.div`
     margin-left: 0;
   }
 `
+
+export const SrOnly = styled.div`
```
Comment (SaraVieira) [inline src/components/Elements.js:115]: This is great! So much better for a11y!

Thank you so much!

File: src/components/Welcome.js [hunk 3]
```diff
@@ -4,13 +4,11 @@ import { WelcomeCard } from './Elements'
...
   <WelcomeCard>
     <div>
-      <h2>Hello stranger!</h2>
-      <h1>So you think you bored!</h1>
+      <h1>Hello stranger!</h1>
+      <h2>So you think you bored!</h2>
       <p>Our chatbot can tell you a cool thing to do. Just give it a try!</p>
     </div>
     <div>
-      <br />
```
Comment (SaraVieira) [inline src/components/Welcome.js:12]: You deleted my shame, thanks 😂

File: src/components/Chat.js [hunk 4]
```diff
@@ -46,26 +47,37 @@ const DateWrapper = styled.span`
...
-        </BotSpeaking>
-      </Fragment>
-    ))}
+const List = styled.ul`
+  padding-left: 0;
+  margin: 0;
+`
+
+const Chat = ({ children, messages }) => {
+  const ariaAttr = (i) => i === messages.length - 1 ? { 'aria-live': 'polite' } : {}
```
Comment (SaraVieira) [inline src/components/Chat.js:56]: I had no idea about `aria-live`!

Dope!

**Review by SaraVieira — APPROVED** [review]
Bless your cotton socks!

File: src/app.js (hunk 1 above)
Comment (sandrina-p) [reply src/app.js:80]: Oh that's cool, I didn't know about that part

File: src/components/Chat.js (hunk 4 above)
Comment (sandrina-p) [reply src/components/Chat.js:56]: aria-live is essential for text that is dynamic, such as chats and notifications.
An [easy to read article](https://developers.google.com/web/fundamentals/accessibility/semantics-aria/hiding-and-updating-content) about it.
The [MDN docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions).

File: src/app.js (hunk 1 above)
Comment (SaraVieira) [reply src/app.js:80]: We all learned something today :D

File: src/components/Chat.js (hunk 4 above)
Comment (SaraVieira) [reply src/components/Chat.js:56]: That is really cool!

I need to be adding this to notifications of errors and stuff

Thank you!

**Pull request merged**